build/common.sh

```diff
@@ -1177,7 +1177,9 @@ function kube::release::gcs::publish() {
 
   local contents
   if [[ ${KUBE_GCS_MAKE_PUBLIC} =~ ^[yY]$ ]]; then
+    kube::log::status "Making uploaded version file public and non-cacheable."
     gsutil acl ch -R -g all:R "${publish_file_dst}" >/dev/null 2>&1 || return 1
+    gsutil setmeta -h "Cache-Control:private, max-age=0" "${publish_file_dst}" >/dev/null 2>&1 || return 1
```

Comment:
https://cloud.google.com/storage/docs/gsutil/commands/setmeta also suggests adding "no-transform", though I don't know if that's strictly necessary.

https://cloud.google.com/storage/docs/gsutil/addlhelp/WorkingWithObjectMetadata says

> Another use of the Cache-Control header is through the “no-transform” value, which instructs Google Cloud Storage to not apply any content transformations based on specifics of a download request, such as removing gzip content-encoding for incompatible clients. Note that this parameter is only respected by the XML API. The Google Cloud Storage JSON API respects only the no-cache and max-age Cache-Control parameters.